allensdk/brain_observatory/behavior/session_apis/data_io/behavior_ophys_nwb_api.py

```diff
@@ -418,86 +473,66 @@ def get_screen_gaze_data(self, include_filtered_data=False) -> Optional[pd.DataF
-        eye_tracking_data['time'] = timestamps
-        eye_tracking_data = pd.DataFrame(eye_tracking_data)
-        eye_tracking_data.index = eye_tracking_data.index.rename('frame')
-        return eye_tracking_data
-
     @staticmethod
-    def _add_eye_tracking_data_to_nwb(nwbfile: NWBFile, eye_tracking: pd.DataFrame, eye_tracking_rig_geometry: dict,
-                                      eye_gaze_mapping_file_path: str, session_sync_file: Path):
+    def add_eye_tracking_data_to_nwb(nwbfile: NWBFile, eye_tracking_df: pd.DataFrame, eye_tracking_rig_geometry: dict,
+                                     eye_gaze_mapping_file_path: Path = None) -> NWBFile:
+        # 1. Add rig geometry
         nwb.add_eye_tracking_rig_geometry_data_to_nwbfile(nwbfile=nwbfile,
                                                           eye_tracking_rig_geometry=eye_tracking_rig_geometry)
 
-        # Collect eye tracking/gaze mapping data from files
-        eye_tracking_frame_times = su.get_synchronized_frame_times(session_sync_file=session_sync_file,
-                                                                   sync_line_label_keys=Dataset.EYE_TRACKING_KEYS)
-
+        # 2. Add eye gaze mapping
         if eye_gaze_mapping_file_path:
             eye_gaze_data = nwb.read_eye_gaze_mappings(Path(eye_gaze_mapping_file_path))
-        else:
-            eye_gaze_data = None
+            nwb.add_eye_gaze_mapping_data_to_nwbfile(nwbfile,
+                                                     eye_gaze_data=eye_gaze_data)
+
+        # 3. Add eye tracking
+        eye_tracking = EllipseSeries(
+            name='eye_tracking',
+            reference_frame='nose',
+            data=eye_tracking_df[['eye_center_x', 'eye_center_y']].values,
+            area=eye_tracking_df['eye_area'].values,
+            width=eye_tracking_df['eye_width'].values,
+            height=eye_tracking_df['eye_height'].values,
+            angle=eye_tracking_df['eye_phi'].values,
+            timestamps=eye_tracking_df['time'].values
+        )
+
+        pupil_tracking = EllipseSeries(
+            name='pupil_tracking',
+            reference_frame='nose',
+            data=eye_tracking_df[['pupil_center_x', 'pupil_center_y']].values,
+            area=eye_tracking_df['pupil_area'].values,
+            width=eye_tracking_df['pupil_width'].values,
+            height=eye_tracking_df['pupil_height'].values,
+            angle=eye_tracking_df['pupil_phi'].values,
+            timestamps=eye_tracking
+        )
+
+        corneal_reflection_tracking = EllipseSeries(
+            name='corneal_reflection_tracking',
+            reference_frame='nose',
+            data=eye_tracking_df[['cr_center_x', 'cr_center_y']].values,
+            area=eye_tracking_df['cr_area'].values,
+            width=eye_tracking_df['cr_width'].values,
+            height=eye_tracking_df['cr_height'].values,
+            angle=eye_tracking_df['cr_phi'].values,
+            timestamps=eye_tracking
```

Comment:
I think you mean this to be an array of timestamps, not the `EllipseSeries` `eye_tracking` instance?